gulpfile.js

```diff
@@ -396,10 +398,94 @@ gulp.task('zip:edge', zipTask('edge'))
 gulp.task('zip:opera', zipTask('opera'))
 gulp.task('zip', gulp.parallel('zip:chrome', 'zip:firefox', 'zip:edge', 'zip:opera'))
 
+
+/*
+ * Brave
+ */
+
+/*
+ * ToDo(ryanml) - Write a method to convert simple paths to a Regex obj
+ * then a simple mapping can be created for these replacements.
+ * Ex (psuedo-ish):
+ * const replacements = [
+ *   {
+ *     target: 'actions',
+ *     replace: 'ui/app/store/actions'
+ *   }
+ * ]
+ *
+ * replacements.map((item) => {
+ *   .pipe(
+ *     replace(convertToRegex(item.target), fmtPath(item.replace))
+ *   )
+ * })
+ */
+
+const overrideDirs = [
+  'ui/app/**/*',
+  'app/scripts/**/*'
+]
+
+const braveRoot = '~/brave/'
+
+gulp.task('dev:override', function () {
+  return gulp.src(overrideDirs)
+    .pipe(
+      replace(
+        /\'(.*)\/home\'/gm,
+        `'${braveRoot}ui/app/pages/home'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'\.\/routes\'/gm,
+        `'${braveRoot}ui/app/pages/routes'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/actions\'/gm,
+        `'${braveRoot}ui/app/store/actions'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/preferences\'/gm,
+        `'${braveRoot}app/scripts/controllers/preferences'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/metamask-controller\'/gm,
+        `'${braveRoot}app/scripts/metamask-controller'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/metamask\/metamask\'/gm,
+        `'${braveRoot}ui/app/ducks/metamask/metamask'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/components\/menu\'/gm,
+        `'${braveRoot}ui/app/components/app/dropdowns/components/menu'`
+      )
+    )
+    .pipe(
+      replace(
+        /\'(.*)\/token-menu-dropdown\.js\'/gm,
+        `'${braveRoot}ui/app/components/app/dropdowns/token-menu-dropdown'`
+      )
+    )
+    .pipe(gulp.dest(file => file.base))
+})
```

Comment:
nit: Pls put this in another file and then patch to import and do it.  Mainly the more we can avoid patches on patches the better.

Reply:
done